logstash-core/spec/logstash/runner_spec.rb

```diff
@@ -18,14 +18,25 @@ def run(args); end
 
   describe "argument parsing" do
 
+    before do
+      #subject.configure # setup_agent requires this to give it logging
+      #subject.setup_agent
+    end
+
     subject { LogStash::Runner.new("") }
     context "when -e is given" do
 
       let(:args) { ["-e", ""] }
+      let(:agent) { double("agent") }
+      let(:agent_logger) { double("agent logger") }
+
+      before do
```

Comment:
I see this pattern shared in most of the test here, I would recommend having explicit before(:each), before(:all), etc statements. I found this syntax more clear to the reader.